docs/src/tut/nft/index.md

```diff
@@ -0,0 +1,739 @@
+1. Itemize the [data definitions](#data-definition) and their primitive types.
+1. Make a list of the participants and how they will [interact](#communication-construction) with one another and the contract. 
+1. Write all [assumptions](#assertion-insertion) about the business logic of the DApp.
+
+Once this information is clarified a flow chart can be created to dictate the organization of the DApp.
+
+## Problem Analysis
+
+Let's begin by defining the problem that this DApp will solve:
+
+1. Collectors of NFTs desire a secondary marketplace to sell their assets, 
+and buyers desire to purchase NFTs that were not previously available.
+1. Sellers must be able to create a new auction for each piece of artwork. 
+1. The NFT must be held by the contract after the auction begins and until the auction closes.
+1. For each bid, the new bid must be higher than the previous bid. 
+1. The previous bid will be refunded to the previous bidder and the new bid will be recorded as the highest bid and be held by the contract.
+1. At the end of a successful auction the highest bidder will receive the NFT and the seller will receive the full bid amount.
+
+## Data Definition
+
+The necessary data points and their types can be determined now that the problem has been analyzed.
+
+In Reach, global constants can be defined using the `const` keyword. 
+Localized backend methods are defined in the Participant Interact Interface, which dictates the DApp's communication model. 
+The methods in each participant's interact interface are reflected as Interact Objects in the frontend. 
+
+Let's define the methods in the Participant Interact Interface:
+
+
+```
+load: /examples/nft-auction-api/index.rsh
+md5: aba879d62803fb298b7ce92187d6a489
+range: 1-17
+```
+
+
+This block initializes the DApp and populates the Participant Interact Interface.
+
++ Line 1: This is the first line in every Reach program and tells the compiler what version of Reach to use
++ Line 3: Initializes the Reach Application
++ Line 4: Instantiates the `Participant` "Creator"
++ Lines 5-9: Defines `getSale`, a function that has no input and outputs an object with three key-value pairs. 
+`nftId` is a `Token`, and `minBid` and `lenInBlocks` are both unsigned integers.
++ Line 10: Creates `auctionReady`, a function that has no inputs or outputs
++ Line 11: Realizes `seeBid`, a function that takes an `Address` and `UInt` as inputs and has no outputs
++ Line 12: The final function, `showOutcome` which also takes an `Address` and `UInt` as inputs and has no output.
++ Line 14: Defines an `API` named `"Bidder"`
++ Line 15: Names `Bidder`'s only function, `bid`, which takes an unsigned integer as an input and outputs a `Tuple` containing an `Address` and `UInt`.
++ Line 17: Initializes the application and begins the first step of the Reach DApp.
+
+Now, let's set up the initial data definitions in the frontend.
+
+
+```
+load: /examples/nft-auction-api/index.mjs
```

Comment:
My guess is that when you send it over to Algorand, you'll need to include the raw code

Reply:
Yes, I'll do that.